lib/saml2.coffee

```diff
@@ -415,12 +450,23 @@ set_option_defaults = (request_options, idp_options, sp_options) ->
 
 module.exports.ServiceProvider =
   class ServiceProvider
-    # Initializes a service provider given the passed options
-    # @entity_id, @private_key, @certificate, @assert_endpoint can only be set here and are used by exported functions.
+    # Initializes a service provider given the passed options.
+    #
+    # @entity_id, @private_key, @assert_endpoint, @certificate, @alt_private_keys, @alt_certs can
+    # only be set here and are used by exported functions.
+    #
     # Rest of options can be set/overwritten by the identity provider and/or at function call.
     constructor: (options) ->
-      {@entity_id, @private_key, @certificate, @assert_endpoint} = options
-      @shared_options = _.pick(options, "force_authn", "auth_context", "nameid_format", "sign_get_request", "allow_unencrypted_assertion")
+      for option in ['entity_id', 'private_key', 'certificate', 'assert_endpoint']
+        throw new Error("#{option} is required") unless options[option]?
+
+      {@entity_id, @private_key, @certificate, @assert_endpoint, @alt_private_keys, @alt_certs} = options
+
+      @alt_private_keys = [].concat(@alt_private_keys or [])
```

Comment:
using this as shorthand for:
```coffeescript
@alt_private_keys = [] unless @alt_private_keys?
@alt_private_keys = [@alt_private_keys] unless _(@alt_private_keys).isArray()
```
can expand it for clarity though, if this isn't very readable